### DISCUSSION TITLE
Compilation fails for specific images

### DISCUSSION THREAD

**Discussion by nameiwillforget** [mrpiggi/svg#57] [Q&A] [answered]
I've been using this package for some time now and never had any issues but now when I use it to include some svg images in a file the compiler complains about misplaced syntax. I'm a bit confounded by this because the compiler doesn't complain about other very similar images and I can't really see what distinguishes some from others. My only possible explanation is that it is perhaps about some of the symbols used in some images. I've included the images for which compilation fails. For the first image, called `layer3.svg`, the compiler says

```
ERROR: Missing } inserted.

--- TeX said ---
<inserted text> 
}
l.89 ...smash{\begin{tabular}[t]{l}}\end{tabular}}}}
                                                  %
```

about the line

```
\put(0.77545014,0.14246878){\color[rgb]{0,0,0}\makebox(0,0)[lt]{\lineheight{1.25}\smash{\begin{tabular}[t]{l}}\end{tabular}}}}%
```

Similarly, for the second image, `DEmacs C-1.svg`, the compiler complains about a missing } bracket in the line

```
\put(0.51608735,0.16528667){\color[rgb]{0,0,0}\makebox(0,0)[t]{\lineheight{1.25}\smash{\begin{tabular}[t]{c}{[\\Cite\\]}\end{tabular}}}}%   
```

and similar for the other images. Compilation happened on Emacs 30.0.5 using LuaLaTeX and a recently upgraded texlive-distribution.

![layer3](https://github.com/mrpiggi/svg/assets/81373487/abea8e76-a8a0-4382-9198-53c630f07f78)
![DEmacs C-1](https://github.com/mrpiggi/svg/assets/81373487/b0ef55d7-59ff-432d-93cc-b4f56e607076)
![DEmacs C-3](https://github.com/mrpiggi/svg/assets/81373487/41eb8d77-9462-457b-aebd-6fdacfbd9276)
![DEmacs C-6](https://github.com/mrpiggi/svg/assets/81373487/544caa03-c560-41a3-ac79-ca43c21cc4ed)

If these images are commented out, compilation succeeds though there are lots of similar images in the document.

**Comment by mrpiggi** [comment]
The code is generated by Inkscape, which is automatically invoked via package svg. And without the SVG files together with a minimal LaTeX document, there is nothing I can do to investigate but only guessing. As the line in question is more or less an empty table, maybe it is worth searching the SVG files for empty text boxes an get rid of them.

**Reply by nameiwillforget** [reply]
Ah, sorry, I thought I was uploading them, but I guess I need to use the other upload button. They should be in the attachments of this comment. 

Edit: I tried uploading the images as files through the "Attach Files" button, but Github made them into images embedded in the text again. How can I just upload them as files?

A minimal TeX document would be this:

```
\documentclass{article}
\usepackage{svg}


\begin{document}

\includesvg{/home/alex/projects/Daselt/pictures/DEmacs C-1}

\end{document}
```
This produces the error

```
ERROR: Missing } inserted.

--- TeX said ---
<inserted text> 
}
l.75 ...n{tabular}[t]{c}{[\\Cite\\]}\end{tabular}}}}                                                  %
```
referring to the line

```
\put(0.51608735,0.16528667){\color[rgb]{0,0,0}\makebox(0,0)[t]{\lineheight{1.25}\smash{\begin{tabular}[t]{c}{[\\Cite\\]}\end{tabular}}}}%
```
I'll look if there's an empty box in the image.

**Reply by nameiwillforget** [reply]
I've looked at the `DEmacs C-1` and `DEmacs C-3` files, and for those two at least the issue seems to be LaTeX-inserted text with curly text brackets that stand on their own or only next to square or other brackets. For instance, this

![DEmacs C-1-test2](https://github.com/mrpiggi/svg/assets/81373487/ff944cad-8e1e-4a4c-8d87-01e9f3fe3d20)

compiles while this

![DEmacs C-1-test](https://github.com/mrpiggi/svg/assets/81373487/b6a43f66-af5e-45ce-a1f0-9a174497f1ec)

does not and neither does this

![DEmacs C-1-test3](https://github.com/mrpiggi/svg/assets/81373487/d3f0511c-3031-4ba7-bfab-da22ee9c80ec)

**Reply by mrpiggi** [reply, marked as answer]
My bad, I've could have downloaded the linked images as those are proper SVG files.

The text within each image is passed to the corresponding `*.pdf_tex` file and directly processed by LaTeX, it is up to you escaping special characters. Compilation succeeds with escaped characters `\{` and `\}` for me.
